Source/CarthageKit/Xcode.swift

```diff
@@ -865,24 +865,29 @@ public func buildScheme(scheme: String, withConfiguration configuration: String,
 	}
 
 	return BuildSettings.SDKsForScheme(scheme, inProject: project)
+		|> map { $0.platform }
 		|> collect
-		|> flatMap(.Concat) { (schemeSDKList: [SDK]) in
-			let platform: Platform! = schemeSDKList.first?.platform
-			
-			if platform == nil {
+		// Exclude duplicated platforms to avoid building the same SDKs multiple
+		// times.
+		|> flatMap(.Concat) { platforms in
+			if platforms.isEmpty {
 				fatalError("No SDKs found for scheme \(scheme)")
 			}
-			
+
+			return SignalProducer(values: Set(platforms))
+		}
+		|> flatMap(.Concat) { (platform: Platform) in
 			let folderURL = workingDirectoryURL.URLByAppendingPathComponent(platform.relativePath, isDirectory: true).URLByResolvingSymlinksInPath!
-			
-			let sdksToBuild = sdkFilter(sdks: schemeSDKList, scheme: scheme, configuration: configuration, project: project)
+
+			let sdksToBuild = sdkFilter(sdks: platform.SDKs, scheme: scheme, configuration: configuration, project: project)
```

Comment:
Can we be sure that this will be equal to the original SDK list? Or should we build a `[Platform:[SDK]]` and get the SDK list from that?